src/web/mjs/connectors/Allanimesite2.mjs

```diff
@@ -0,0 +1,102 @@
+import Allanimesite from './Allanimesite.mjs';
+export default class Allanimesite2 extends Allanimesite {
+    constructor() {
+        super();
+        super.id = 'allanimesite2';
+        super.label = 'AllAnime.site (Animes)';
+        this.tags = ['anime', 'multi-lingual'];
+        this.url = 'https://allanime.site';
+        this.path = '/allanimeapi';
+        this.varQueryMangas = '?variables={"search":{"allowAdult":true,"allowUnknown":true},"limit":100,"page":%PAGE%,"countryOrigin":"ALL"}&extensions={"persistedQuery":{"version":1,"sha256Hash":"9c7a8bc1e095a34f2972699e8105f7aaf9082c6e1ccd56eab99c2f1a971152c6"}}';
+    }
+    get icon() {
+        return '/img/connectors/allanimesite';
+    }
+    canHandleURI(uri) {
+        return /(www\.)?allanime\.site\/anime/.test(uri);
+    }
+    async _getMangasFromPage(page) {
+        const uri = new URL(this.path + this.varQueryMangas.replace('%PAGE%', page), this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        if (data.data == null) {

```

Comment:
I usually use !data.data (but it also includes false, zero and empty strings), just do as you want ;)
You can also do an inline if
`if (!data.data) return {};`
also, isn't it supposed to return an array ? []